build_all.sh

```diff
@@ -33,7 +33,8 @@ build_nim_csources(){
       if [ "$unamestr" = 'FreeBSD' ]; then
         makeX=gmake
       fi
-      which $makeX && echo_run $makeX -C csources -j -l 80 || build_nim_csources_via_script
+      nCPU=$(nproc 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || getconf _NPROCESSORS_ONLN 2>/dev/null || 1)
+      which $makeX && echo_run $makeX -C csources -j -l $nCPU || build_nim_csources_via_script
```

Comment:
Please don't use the `-j` flag alone. While load limiting will eventually limit the spawn count, on system with slower disk access (for example an old 5k4rpm hdd, like on my laptop), it will saturate the pid count and system resources before the C compiler actually start compiling (since the tons of spawned processes initialize too slow, and `make` don't terminate spawned processes).

Reply:
* so... we're back to what I had in https://github.com/nim-lang/Nim/pull/13300 (but maybe the change `[[ $# -ne 0 ]]` to `[ $# -ne 0 ]` is good)
* @alaviss can you give timing results with `-j nproc -l nproc` vs `-j -l nproc` on your system (taking care of only timing the make command, and taking care of starting from a fresh csources clone)

Reply:
`make -j 4 -l 4`: real 1m7.722s user 4m5.971s sys 0m5.769s
`make -j -l 4`: real 1m18.001s user 3m44.192s sys 0m5.329s
`make -j 4`: real 1m7.675s user 4m5.909s sys 0m5.663s

The fs cache has been pre-populated with 3 rounds of `find -type f -exec cat '{}' + > /dev/null` before the test series begin. I'm on ZFS with a big ARC cache so the caching is rather aggressive.

Fun fact, once the cache is dropped, `make -j -l 4` hit me with an 7.10 load before hovering over 6.98. While the time difference is rather minimal, it certainly hits many other app that I was running at the same time.

Did I also mention that I'm on a kernel with MuQSS scheduler which is better at scheduling overloaded scenarios than the mainline CFS (it favors interactivity over throughput)?

Reply:
Ok, addressed